### ISSUE TITLE
NVDA incorrectly identifies selected cells in Excel when selection is non-contiguous

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#7177]
In Excel, when a selection is non-contiguous, NVDA incorrectly identifies the selected cells.

Non-contiguous cells can be selected by using the "Go to special" feature (F5, then alt+s) to select all cells meeting a particular requirement - eg all cells with constants, or all cells with comments.

To reproduce:
Type the number 1 in cells A1, B2 and C2.
Press F5, then alt+s.  Down arrow once to select "Constants" and press ENTER.
Expected behaviour: NVDA should announce "Selected, A1, B2 through C2" (or similar).
Actual behaviour: NVDA announces: "Selected A1 through A3".

**Comment by SaschaCowley** [comment]
Tested with NVDA version alpha-37521,f29d41fb (2025.3.0.37521) and Microsoft® Excel® for Microsoft 365 MSO (Version 2506 Build 16.0.18925.20076) 64-bit. I also put "1" in cells D3, D4, E3 and E4.

* With Use UI Automation to access Microsoft Excel spreadsheet controls when available off (default), nothing is reported after pressing enter in the Go To Special dialog.
* With it on, NVDA reports "selected  D3 1 through E4 1". As far as I can tell (I tested various other combinations), it is reporting the lower right-most selection rectangle.
